### ISSUE TITLE
PointCloud::RemoveStatisticalOutliers prints progress bar even when print_progress = false

### ISSUE THREAD

**Issue by franciscorba** [isl-org/Open3D#5599]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### Describe the issue

`PointCloud::RemoveStatisticalOutliers` is still printing the progress bar even when the argument `print_progress` is set to `false`.

### Steps to reproduce the bug

```python
Given a `PointCloud` `pc`:


pc->RemoveStatisticalOutliers(50, 1.0, false);
```

### Error message

```
Remove statistical outliers: [=>                          
Remove statistical outliers: [========================================] 100%

### Expected behavior

No output

### Open3D, Python and System information

```markdown
- Operating system: Ubuntu 18.04
- Python version: Python 3.6.9
- Open3D version: 0.14.1+4dea5bebb
- System architecture: x86
- Is this a remote workstation?: no
- How did you install Open3D?: build from source
- Compiler version (if built from source): gcc 7.5.0
```

**Comment by franciscorba** [comment]
Checking the `PointCloud.cpp` file I found in line 551-552 that `print_progress` isn't even been passed to the progress bar:

```
utility::OMPProgressBar progress_bar(points_.size(),
                                         "Remove statistical outliers: ", true);

**Comment by franciscorba** [comment]
This was solved and is no more present on version 0.15.0.
